### PR TITLE
issue #69 Add kata bowling

### LINKED ISSUE #69: [TDD] Kata - Bowling
**Description**
To calculate bowling game scores.
[Kata-bowling](https://codingdojo.org/kata/Bowling/)

We can briefly summarize the scoring for this form of bowling:

Each game, or “line” of bowling, includes ten turns, or “frames” for the bowler.

- In each frame, the bowler gets up to two tries to knock down all the pins.
- If in two tries, he fails to knock them all down, his score for that frame is the total number of pins knocked down in his two tries.
- If in two tries he knocks them all down, this is called a “spare” and his score for the frame is ten plus the number of pins knocked down on his next throw (in his next turn).
- If on his first try in the frame he knocks down all the pins, this is called a “strike”. His turn is over, and his score for the frame is ten plus the simple total of the pins knocked down in his next two rolls.
- If he **gets a spare or strike in the last (tenth) frame, the bowler gets to throw one or two more bonus balls**, respectively. These bonus throws are taken as part of the same turn. If the bonus throws knock down all the pins, the process does not repeat: the bonus throws are only used to calculate the score of the final frame.
- The game score is the total of all frame scores.

```
Given one parameter (string array)
When input as X X X X X X X X X X X X (12 rolls: 12 strikes)
Expect return 300 (10 frames * 30 points)

When input as 9- 9- 9- 9- 9- 9- 9- 9- 9- 9- (20 rolls: 10 pairs of 9 and miss) 
Expect return 90 (10 frames * 9 points)

When input as 5/ 5/ 5/ 5/ 5/ 5/ 5/ 5/ 5/ 5/5 (21 rolls: 10 pairs of 5 and spare, with a final 5)
Expect return 150 (10 frames * 15 points )

```
**Additional context**

Add any other context, links references or screenshots about the feature request here.

**Important Checklist**
- [ ] I have written unit test first, make it red(fail first).
- [ ] I have then written logic code, make it green(success)
- [ ] I have tried to refactor my code, make it understandable(document my code)

### PR DESCRIPTION
**Pull request by farzado**
Add kata bowling

* Add kata bowling

Resolves #69

### CODE REVIEW

**Comment by farzado** [comment]
@fengyuanyang Could you pls review this one too? Appreciated.

**Comment by farzado** [comment]
@fengyuanyang @januslinhc Just in case you missed the notification, this PR is ready for review.

**Comment by fengyuanyang** [comment]
Hi @farzado 
I did review yours here, can you see it ?

**Review by fengyuanyang — COMMENTED** [review]
Hi @farzado 
can you see my comments?

**Comment by farzado** [comment]
> Hi @farzado
> I did review yours here, can you see it ?

Thanks @fengyuanyang, I can see it only now. Will update the PR soon.

File: src/main/java/com/ordestiny/tdd/kata/BowlingGame.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      separateScores = getSeparateRollScores(currentFrameScore);
+      if (prevScoreType == ScoreType.SPARE) {
+        interimScores[currentFrameIndex - 1] += separateScores[0];
+        if (currentFrameScore.contains(STRIKE)) {
+          interimScores[currentFrameIndex - 1] += 10;
+        }
+      }
+      if (prevScoreType == ScoreType.STRIKE) {
+        interimScores[currentFrameIndex - 1] += separateScores[0] + separateScores[1];
+      }
```
[the remark this answers is not in the document]
Comment (farzado) [reply src/main/java/com/ordestiny/tdd/kata/BowlingGame.java]: Thanks for your suggestion. I've updated the PR.

**Comment by farzado** [comment]
@fengyuanyang The PR is now updated and contains the change you suggested.

**Comment by fengyuanyang** [comment]
> @fengyuanyang The PR is now updated and contains the change you suggested.

Thanks , looks much better , and much easier to understand 👍

**Pull request merged**